src/components/editor/Controls.tsx

```diff
@@ -11,25 +12,56 @@ const Label = ({ text, children }: {
 	</label>
 }
 
-export default function Controls({ exportHandler }: {
-	exportHandler: (data: { size: number, fps: number, speed: number, loop_gif: boolean, bounce: boolean }) => void,
+const getEstimatedFileSize = (
+	fps: number,
+	width: number,
+	height: number,
+	quality: number,
+	durationInFrames: number
+) => {
+	const durationInSeconds = durationInFrames / CONSTANTS.previewFps;
+	const totalFrames = durationInSeconds * fps;
+	const qFactor = quality / 100;
+
+	let totalPixels = (width * height * totalFrames) / 3;
+
+	let totalBytes = (totalPixels * qFactor + 1.5) / 2.5
+	const totalKb = totalBytes / 1024;
+	const totalMb = totalKb / 1024;
+	return totalMb.toFixed(2);
+}
+
+export default function Controls({
+	exportHandler,
+	selectedFrames,
+}: {
+	exportHandler: (data: {
+		fps: number,
+		size: number,
+		speed: number,
+		bounce: boolean,
+		loop_gif: boolean
+	}) => void,
+	selectedFrames: number[],
 	exporting: boolean
 }) {
 
-	const [size, setSize] = useState(1000);
 	const [fps, setFps] = useState(30);
-	const [speed, setSpeed] = useState(1);
+	const [size, setSize] = useState(1000);
 	const [loop, setLoop] = useState(false);
+	const [speed, setSpeed] = useState(1);
 	const [bounce, setBounce] = useState(false);
 
+	const totalFrames = Math.abs(selectedFrames[1] - selectedFrames[0]);
+
+	const estimatedSize = getEstimatedFileSize(fps, totalFrames, size, size, 90);
```

Comment:
I think I don't understand the function 😅  but doesn't 
`getEstimatedFileSize` expect these arguments?
`
	fps: number,
	width: number,
	height: number,
	quality: number,
	durationInFrames: number
`
Are we passing the right arguments? What's the significance of the magic `90` ?

Reply:
Yup, you are correct. Added proper variable names and explained the magic 90 😅